version.semver.json

```diff
@@ -0,0 +1,15 @@
+{
+    "semver": {
```

Comment:
I may very well be asking a dumb question, but why so much complexity? Why not simply `1.4.1.-rc.1`?

Reply:
My goal is not to make it complex, but explicit. The idea is to provide a machine readable way to reason about the version information.

As you know, a tag is sufficiently unique to work with our external tools as they are today, and indeed the `print-version.sh` script is intended to generate the such a tag, `1.4.1-rc.1`.

I elected to go this route, so our tools can evolve to have more complex reasoning around version information _(i.e. testing implicit API compatibility guaranteed by the SemVer notation)_. Since this information is versioned, it is guaranteed to be available and easily parsable by future tools from this point forward.